tests/integration/UploadSlideBackTranslationIntegrationTest.php

```diff
@@ -0,0 +1,267 @@
+    private static Process $process;
+    private static string $filesRoot;
+    private static string $uploadedProjectDir;
+    private static Filesystem $fileSystem;
+
+    private Client $httpClient;
+
+    public static function setUpBeforeClass(): void
+    {
+        // start php in-built server
+        self::$process = new Process(["php", "-S", self::HOST, "-t", ROOT_PATH . "."]);
+        self::$process->start();
+        usleep(100000);
+
+        // create required directory for test data
+        self::$fileSystem = new Filesystem();
+        self::$filesRoot = $GLOBALS['filesRoot'];
+        $templateDir = self::$filesRoot . "/Templates";
+        self::$fileSystem->mkdir($templateDir);
+        // copy test story template files to files root
+        self::$fileSystem->mirror(dirname(__DIR__) . "/data/templates", $templateDir);
+        self::$uploadedProjectDir = self::$filesRoot . "/Projects";
+
+        self::$db = new PDO(DB_DNS,
+            $GLOBALS['databaseUser'], $GLOBALS['databasePassword']);
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        self::$fileSystem->remove(self::$filesRoot);
+        self::$process->stop();
+        self::$db->query('DELETE FROM Slide');
+        self::$db->query('DELETE FROM Stories');
+        self::$db = null;
+    }
+
+    public function setUp(): void
+    {
+        $this->httpClient = new Client(['http_errors' => false]);
+    }
+
+    /**
+     * @return int
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Exception
+     */
+    public function testCreateNewStory(): void
+    {
+        // request payload
+        $payload = [
+            'Key' => '',
+            'PhoneId' => self::PHONE_ID_1,
+            'Data' => base64_encode(self::STORY_FILE_CONTENTS),
+            'TemplateTitle' => self::STORY_TEMPLATE,
+            'IsWholeStory' => "true"
+        ];
+
+        $storyId = $this->sendRequestAndReturnStoryId($payload);
+
+        # verify story is created correctly in the database
+        $stories = $this->getStories($storyId);
+        $this->assertCount(1, $stories);
+        $this->assertEquals(self::STORY_TEMPLATE, $stories[0]['title']);
+        $this->assertEquals($this->getProjectId($payload['PhoneId']), $stories[0]['projectId']);
+
+        # verify slides are all created correctly in the database
+        $slides = $this->getSlides($storyId);
+        $this->assertCount(3, $slides);
+        $this->assertEquals(0, $slides[0]['slideNumber']);
+        $this->assertEquals(1, $slides[1]['slideNumber']);
+        # In test `story.json`, third slide type is `COPYRIGHT` which is skipped as per implementation
+        $this->assertEquals(3, $slides[2]['slideNumber']);
+
+        $storyFile = sprintf("%s/%s/%s/wholeStory.m4a", self::$uploadedProjectDir, $payload['PhoneId'], $storyId);
+        $this->assertFileExists($storyFile);
+        $this->assertEquals(file_get_contents($storyFile), base64_decode($payload['Data']));
```

Comment:
Can you change this to `$this->assertEquals(file_get_contents($storyFile), self::STORY_FILE_CONTENTS);` so that there's less logic going on in the test?
Here and below.